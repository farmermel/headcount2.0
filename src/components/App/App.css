body {
  background-color: #f1f7ef;
  font-family: 'Open Sans', sans-serif;
}

