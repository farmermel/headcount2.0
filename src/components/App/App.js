import React, { Component } from 'react';
import CardContainer from '../CardContainer/CardContainer';
import Header from "../Header/Header";
import DistrictRepository from '../../helper';
import kinderData from '../../data/kindergartners_in_full_day_program';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      districts: []
    };
  }

  componentDidMount() {
    const districts = new DistrictRepository(kinderData).findAllMatches();
    
    this.setState({ districts });
  }

  render() {
    return (
      <div>
        <Header />
        <CardContainer districts={this.state.districts} />
      </div>
    );
  }
}

export default App;
