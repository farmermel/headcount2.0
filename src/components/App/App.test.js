import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import DistrictRepository, { mockDistrictRepository } from '../../helper';

jest.mock('../../helper');

describe('App', () => {
  let wrapper;
  let defaultState;

  beforeEach(() => {
    wrapper = shallow(<App />);
    defaultState = {
      districts: [],
      districtRepository: new mockDistrictRepository(),
      compare: []
    };
    // DistrictRepository.mockClear();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    //why is it setting state to districts if componentdidmount isn't running?
    wrapper.setState(defaultState);

    expect(wrapper.state()).toEqual(defaultState);
    expect(wrapper.state().districts.length).toEqual(0);
  });

  it('should update state with districts data when component mounts', () => {
    const wrapper = mount(<App />);

    expect(typeof wrapper.state().districts).toEqual('object');
    expect(wrapper.state().districts.length).toEqual(2);
  });

  describe('searchDistricts', () => {

    it('should make a call to DistrictRepository findByName method', () => {
      expect(wrapper.state().districtRepository.findAllMatches).toHaveBeenCalled();

      wrapper.instance().searchDistricts('COLORADO');

      expect(wrapper.state().districtRepository.findAllMatches).toHaveBeenCalledWith('COLORADO');
    })

    it('should update state with districts found in search', () => {
      expect(wrapper.state().districts.length).toEqual(2);
      expect(wrapper.state().districts[0].location).toEqual('COLORADO');
      expect(wrapper.state().districts[1].location).toEqual('ASPEN 20');

      wrapper.instance().searchDistricts('COLORADO');

      expect(wrapper.state().districts.length).toEqual(1);
      expect(wrapper.state().districts[0].location).toEqual('COLORADO');
    })
    
  })
});
