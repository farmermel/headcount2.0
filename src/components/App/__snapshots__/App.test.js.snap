// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          searchDistricts={[Function]}
/>,
        <ComparisonContainer
          comparativeAnalysis={Object {}}
          compare={[Function]}
          districts={Object {}}
/>,
        <CardContainer
          compare={[Function]}
          comparison={Object {}}
          districts={
                    Array [
                              Object {
                                "data": Object {
                                  "2003": 0.4,
                                  "2004": 0.8,
                                },
                                "location": "COLORADO",
                              },
                              Object {
                                "data": Object {
                                  "2003": 0.7,
                                  "2004": 0.9,
                                },
                                "location": "ASPEN 20",
                              },
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchDistricts": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparativeAnalysis": Object {},
          "compare": [Function],
          "districts": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "compare": [Function],
          "comparison": Object {},
          "districts": Array [
            Object {
              "data": Object {
                "2003": 0.4,
                "2004": 0.8,
              },
              "location": "COLORADO",
            },
            Object {
              "data": Object {
                "2003": 0.7,
                "2004": 0.9,
              },
              "location": "ASPEN 20",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            searchDistricts={[Function]}
/>,
          <ComparisonContainer
            comparativeAnalysis={Object {}}
            compare={[Function]}
            districts={Object {}}
/>,
          <CardContainer
            compare={[Function]}
            comparison={Object {}}
            districts={
                        Array [
                                    Object {
                                      "data": Object {
                                        "2003": 0.4,
                                        "2004": 0.8,
                                      },
                                      "location": "COLORADO",
                                    },
                                    Object {
                                      "data": Object {
                                        "2003": 0.7,
                                        "2004": 0.9,
                                      },
                                      "location": "ASPEN 20",
                                    },
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchDistricts": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparativeAnalysis": Object {},
            "compare": [Function],
            "districts": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "compare": [Function],
            "comparison": Object {},
            "districts": Array [
              Object {
                "data": Object {
                  "2003": 0.4,
                  "2004": 0.8,
                },
                "location": "COLORADO",
              },
              Object {
                "data": Object {
                  "2003": 0.7,
                  "2004": 0.9,
                },
                "location": "ASPEN 20",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
